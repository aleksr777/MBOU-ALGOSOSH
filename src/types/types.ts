import LinkedList from '../components/list-page/linked-list-class'

export type OnlyStringObjType = {
  [ key: string ]: string
}

export type OnlyNumberObjType = {
  [ key: string ]: number
}

export type ButtonState = {
  isLoading: boolean
}

export type ButtonsHookState = {
  [ key: string ]: ButtonState
}

export type StateIndeces = {
  changing: null | [ number, number ]
  modified: null | [ number, number ]
}

export type ListType = LinkedList<unknown | string>

export type ArrChangingType = [ number | null, number | null ]

export type ArrSortingType = ( number | number[] | ArrChangingType )[]

export type StepsSortingType = ArrSortingType[]
