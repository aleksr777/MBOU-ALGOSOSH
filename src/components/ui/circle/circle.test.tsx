import React from 'react'
import { render } from '@testing-library/react'
import Circle from './circle'
import { ElementStates } from '../../../types/element-states'

describe( '<Circle />', () => {

  it( 'render without letter', () => {
    const { asFragment } = render( <Circle /> )
    expect( asFragment() ).toMatchSnapshot()
  } )

  it( 'render with letters', () => {
    const { asFragment } = render( <Circle letter="A" /> )
    expect( asFragment() ).toMatchSnapshot()
  } )

  it( 'render with head', () => {
    const { asFragment } = render( <Circle head="Head Content" /> )
    expect( asFragment() ).toMatchSnapshot()
  } )

  it( 'render correctly with react element in head', () => {
    const { asFragment } = render(
      <Circle head={
        <Circle letter={ 'A' } state={ ElementStates.Changing } isSmall={ true } />
      } /> )
    expect( asFragment() ).toMatchSnapshot()
  } )

  it( 'render correctly with tail', () => {
    const { asFragment } = render( <Circle tail="Tail Content" /> )
    expect( asFragment() ).toMatchSnapshot()
  } )

  it( 'render correctly with react element in tail', () => {
    const { asFragment } = render( <Circle tail={
      <Circle letter={ 'A' } state={ ElementStates.Changing } isSmall={ true } />
    } /> )
    expect( asFragment() ).toMatchSnapshot()
  } )

  it( 'render correctly with index', () => {
    const { asFragment } = render( <Circle index={ 5 } /> )
    expect( asFragment() ).toMatchSnapshot()
  } )

  it( 'render correctly with isSmall prop', () => {
    const { asFragment } = render( <Circle isSmall={ true } /> )
    expect( asFragment() ).toMatchSnapshot()
  } )

  it( 'render correctly in default state', () => {
    const { asFragment } = render( <Circle state={ ElementStates.Default } /> )
    expect( asFragment() ).toMatchSnapshot()
  } )

  it( 'renders correctly in changing state', () => {
    const { asFragment } = render( <Circle state={ ElementStates.Changing } /> )
    expect( asFragment() ).toMatchSnapshot()
  } )

  it( 'render correctly in modified state', () => {
    const { asFragment } = render( <Circle state={ ElementStates.Modified } /> )
    expect( asFragment() ).toMatchSnapshot()
  } )
} )
