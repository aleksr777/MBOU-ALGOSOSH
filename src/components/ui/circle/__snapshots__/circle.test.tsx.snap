// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Circle /> render correctly in default state 1`] = `
<DocumentFragment>
  <div
    class="content "
  >
    <div
      class="text text_type_input text_color_input mb-4 absolute head element"
    />
    <div
      class="circle   default"
    >
      <p
        class="text text_type_circle text_color_input letter"
      />
    </div>
    <p
      class="text text_type_input text_color_input mt-4 absolute index"
    />
    <div
      class="text text_type_input text_color_input mt-4 absolute tail30 element"
    />
  </div>
</DocumentFragment>
`;

exports[`<Circle /> render correctly in modified state 1`] = `
<DocumentFragment>
  <div
    class="content "
  >
    <div
      class="text text_type_input text_color_input mb-4 absolute head element"
    />
    <div
      class="circle   modified"
    >
      <p
        class="text text_type_circle text_color_input letter"
      />
    </div>
    <p
      class="text text_type_input text_color_input mt-4 absolute index"
    />
    <div
      class="text text_type_input text_color_input mt-4 absolute tail30 element"
    />
  </div>
</DocumentFragment>
`;

exports[`<Circle /> render correctly with index 1`] = `
<DocumentFragment>
  <div
    class="content "
  >
    <div
      class="text text_type_input text_color_input mb-4 absolute head element"
    />
    <div
      class="circle   default"
    >
      <p
        class="text text_type_circle text_color_input letter"
      />
    </div>
    <p
      class="text text_type_input text_color_input mt-4 absolute index"
    >
      5
    </p>
    <div
      class="text text_type_input text_color_input mt-4 absolute tail60 element"
    />
  </div>
</DocumentFragment>
`;

exports[`<Circle /> render correctly with isSmall prop 1`] = `
<DocumentFragment>
  <div
    class="content "
  >
    <div
      class="text text_type_input text_color_input mb-4 absolute head element"
    />
    <div
      class="circle  small default"
    >
      <p
        class="text text_type_circle text_color_input letter"
      />
    </div>
    <p
      class="text text_type_input text_color_input mt-4 absolute index"
    />
    <div
      class="text text_type_input text_color_input mt-4 absolute tail30 element"
    />
  </div>
</DocumentFragment>
`;

exports[`<Circle /> render correctly with react element in head 1`] = `
<DocumentFragment>
  <div
    class="content "
  >
    <div
      class="text text_type_input text_color_input mb-4 absolute head element"
    >
      <div
        class="content "
      >
        <div
          class="text text_type_input text_color_input mb-4 absolute head element"
        />
        <div
          class="circle  small changing"
        >
          <p
            class="text text_type_circle text_color_input letter"
          >
            A
          </p>
        </div>
        <p
          class="text text_type_input text_color_input mt-4 absolute index"
        />
        <div
          class="text text_type_input text_color_input mt-4 absolute tail30 element"
        />
      </div>
    </div>
    <div
      class="circle   default"
    >
      <p
        class="text text_type_circle text_color_input letter"
      />
    </div>
    <p
      class="text text_type_input text_color_input mt-4 absolute index"
    />
    <div
      class="text text_type_input text_color_input mt-4 absolute tail30 element"
    />
  </div>
</DocumentFragment>
`;

exports[`<Circle /> render correctly with react element in tail 1`] = `
<DocumentFragment>
  <div
    class="content "
  >
    <div
      class="text text_type_input text_color_input mb-4 absolute head element"
    />
    <div
      class="circle   default"
    >
      <p
        class="text text_type_circle text_color_input letter"
      />
    </div>
    <p
      class="text text_type_input text_color_input mt-4 absolute index"
    />
    <div
      class="text text_type_input text_color_input mt-4 absolute tail30 element"
    >
      <div
        class="content "
      >
        <div
          class="text text_type_input text_color_input mb-4 absolute head element"
        />
        <div
          class="circle  small changing"
        >
          <p
            class="text text_type_circle text_color_input letter"
          >
            A
          </p>
        </div>
        <p
          class="text text_type_input text_color_input mt-4 absolute index"
        />
        <div
          class="text text_type_input text_color_input mt-4 absolute tail30 element"
        />
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`<Circle /> render correctly with tail 1`] = `
<DocumentFragment>
  <div
    class="content "
  >
    <div
      class="text text_type_input text_color_input mb-4 absolute head element"
    />
    <div
      class="circle   default"
    >
      <p
        class="text text_type_circle text_color_input letter"
      />
    </div>
    <p
      class="text text_type_input text_color_input mt-4 absolute index"
    />
    <div
      class="text text_type_input text_color_input mt-4 absolute tail30 string"
    >
      Tail Content
    </div>
  </div>
</DocumentFragment>
`;

exports[`<Circle /> render with head 1`] = `
<DocumentFragment>
  <div
    class="content "
  >
    <div
      class="text text_type_input text_color_input mb-4 absolute head string"
    >
      Head Content
    </div>
    <div
      class="circle   default"
    >
      <p
        class="text text_type_circle text_color_input letter"
      />
    </div>
    <p
      class="text text_type_input text_color_input mt-4 absolute index"
    />
    <div
      class="text text_type_input text_color_input mt-4 absolute tail30 element"
    />
  </div>
</DocumentFragment>
`;

exports[`<Circle /> render with letters 1`] = `
<DocumentFragment>
  <div
    class="content "
  >
    <div
      class="text text_type_input text_color_input mb-4 absolute head element"
    />
    <div
      class="circle   default"
    >
      <p
        class="text text_type_circle text_color_input letter"
      >
        A
      </p>
    </div>
    <p
      class="text text_type_input text_color_input mt-4 absolute index"
    />
    <div
      class="text text_type_input text_color_input mt-4 absolute tail30 element"
    />
  </div>
</DocumentFragment>
`;

exports[`<Circle /> render without letter 1`] = `
<DocumentFragment>
  <div
    class="content "
  >
    <div
      class="text text_type_input text_color_input mb-4 absolute head element"
    />
    <div
      class="circle   default"
    >
      <p
        class="text text_type_circle text_color_input letter"
      />
    </div>
    <p
      class="text text_type_input text_color_input mt-4 absolute index"
    />
    <div
      class="text text_type_input text_color_input mt-4 absolute tail30 element"
    />
  </div>
</DocumentFragment>
`;

exports[`<Circle /> renders correctly in changing state 1`] = `
<DocumentFragment>
  <div
    class="content "
  >
    <div
      class="text text_type_input text_color_input mb-4 absolute head element"
    />
    <div
      class="circle   changing"
    >
      <p
        class="text text_type_circle text_color_input letter"
      />
    </div>
    <p
      class="text text_type_input text_color_input mt-4 absolute index"
    />
    <div
      class="text text_type_input text_color_input mt-4 absolute tail30 element"
    />
  </div>
</DocumentFragment>
`;

exports[`<Circle /> renders correctly in default state 1`] = `
<DocumentFragment>
  <div
    class="content "
  >
    <div
      class="text text_type_input text_color_input mb-4 absolute head element"
    />
    <div
      class="circle   default"
    >
      <p
        class="text text_type_circle text_color_input letter"
      />
    </div>
    <p
      class="text text_type_input text_color_input mt-4 absolute index"
    />
    <div
      class="text text_type_input text_color_input mt-4 absolute tail30 element"
    />
  </div>
</DocumentFragment>
`;

exports[`<Circle /> renders correctly in modified state 1`] = `
<DocumentFragment>
  <div
    class="content "
  >
    <div
      class="text text_type_input text_color_input mb-4 absolute head element"
    />
    <div
      class="circle   modified"
    >
      <p
        class="text text_type_circle text_color_input letter"
      />
    </div>
    <p
      class="text text_type_input text_color_input mt-4 absolute index"
    />
    <div
      class="text text_type_input text_color_input mt-4 absolute tail30 element"
    />
  </div>
</DocumentFragment>
`;

exports[`<Circle /> renders correctly with head 1`] = `
<DocumentFragment>
  <div
    class="content "
  >
    <div
      class="text text_type_input text_color_input mb-4 absolute head string"
    >
      Head Content
    </div>
    <div
      class="circle   default"
    >
      <p
        class="text text_type_circle text_color_input letter"
      />
    </div>
    <p
      class="text text_type_input text_color_input mt-4 absolute index"
    />
    <div
      class="text text_type_input text_color_input mt-4 absolute tail30 element"
    />
  </div>
</DocumentFragment>
`;

exports[`<Circle /> renders correctly with index 1`] = `
<DocumentFragment>
  <div
    class="content "
  >
    <div
      class="text text_type_input text_color_input mb-4 absolute head element"
    />
    <div
      class="circle   default"
    >
      <p
        class="text text_type_circle text_color_input letter"
      />
    </div>
    <p
      class="text text_type_input text_color_input mt-4 absolute index"
    >
      5
    </p>
    <div
      class="text text_type_input text_color_input mt-4 absolute tail60 element"
    />
  </div>
</DocumentFragment>
`;

exports[`<Circle /> renders correctly with isSmall prop 1`] = `
<DocumentFragment>
  <div
    class="content "
  >
    <div
      class="text text_type_input text_color_input mb-4 absolute head element"
    />
    <div
      class="circle  small default"
    >
      <p
        class="text text_type_circle text_color_input letter"
      />
    </div>
    <p
      class="text text_type_input text_color_input mt-4 absolute index"
    />
    <div
      class="text text_type_input text_color_input mt-4 absolute tail30 element"
    />
  </div>
</DocumentFragment>
`;

exports[`<Circle /> renders correctly with letters 1`] = `
<DocumentFragment>
  <div
    class="content "
  >
    <div
      class="text text_type_input text_color_input mb-4 absolute head element"
    />
    <div
      class="circle   default"
    >
      <p
        class="text text_type_circle text_color_input letter"
      >
        A
      </p>
    </div>
    <p
      class="text text_type_input text_color_input mt-4 absolute index"
    />
    <div
      class="text text_type_input text_color_input mt-4 absolute tail30 element"
    />
  </div>
</DocumentFragment>
`;

exports[`<Circle /> renders correctly with react element in head 1`] = `
<DocumentFragment>
  <div
    class="content "
  >
    <div
      class="text text_type_input text_color_input mb-4 absolute head element"
    >
      <div
        class="content "
      >
        <div
          class="text text_type_input text_color_input mb-4 absolute head element"
        />
        <div
          class="circle  small changing"
        >
          <p
            class="text text_type_circle text_color_input letter"
          >
            A
          </p>
        </div>
        <p
          class="text text_type_input text_color_input mt-4 absolute index"
        />
        <div
          class="text text_type_input text_color_input mt-4 absolute tail30 element"
        />
      </div>
    </div>
    <div
      class="circle   default"
    >
      <p
        class="text text_type_circle text_color_input letter"
      />
    </div>
    <p
      class="text text_type_input text_color_input mt-4 absolute index"
    />
    <div
      class="text text_type_input text_color_input mt-4 absolute tail30 element"
    />
  </div>
</DocumentFragment>
`;

exports[`<Circle /> renders correctly with react element in tail 1`] = `
<DocumentFragment>
  <div
    class="content "
  >
    <div
      class="text text_type_input text_color_input mb-4 absolute head element"
    />
    <div
      class="circle   default"
    >
      <p
        class="text text_type_circle text_color_input letter"
      />
    </div>
    <p
      class="text text_type_input text_color_input mt-4 absolute index"
    />
    <div
      class="text text_type_input text_color_input mt-4 absolute tail30 element"
    >
      <div
        class="content "
      >
        <div
          class="text text_type_input text_color_input mb-4 absolute head element"
        />
        <div
          class="circle  small changing"
        >
          <p
            class="text text_type_circle text_color_input letter"
          >
            A
          </p>
        </div>
        <p
          class="text text_type_input text_color_input mt-4 absolute index"
        />
        <div
          class="text text_type_input text_color_input mt-4 absolute tail30 element"
        />
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`<Circle /> renders correctly with tail 1`] = `
<DocumentFragment>
  <div
    class="content "
  >
    <div
      class="text text_type_input text_color_input mb-4 absolute head element"
    />
    <div
      class="circle   default"
    >
      <p
        class="text text_type_circle text_color_input letter"
      />
    </div>
    <p
      class="text text_type_input text_color_input mt-4 absolute index"
    />
    <div
      class="text text_type_input text_color_input mt-4 absolute tail30 string"
    >
      Tail Content
    </div>
  </div>
</DocumentFragment>
`;

exports[`<Circle /> renders correctly without letter 1`] = `
<DocumentFragment>
  <div
    class="content "
  >
    <div
      class="text text_type_input text_color_input mb-4 absolute head element"
    />
    <div
      class="circle   default"
    >
      <p
        class="text text_type_circle text_color_input letter"
      />
    </div>
    <p
      class="text text_type_input text_color_input mt-4 absolute index"
    />
    <div
      class="text text_type_input text_color_input mt-4 absolute tail30 element"
    />
  </div>
</DocumentFragment>
`;
